src/logging.cpp

```diff
@@ -343,7 +320,12 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
         std::string s{"["};
 
         if (category != LogFlags::NONE) {
-            s += LogCategoryToStr(category);
+            auto it = LogCategoryToStr.find(category);
```

Comment:
Find on a std::unordered set has constant-to-linear (size of the container) complexity, whereas if you build a std::array instead, access with `[]` or `.at` will be constant time (and the array can be constexpr).

Reply:
I am not really sure how to do that.

`LogCategory` enum values are bit flags (ie 1 << i) with the exception of `None` (`0`) and `All`(`~0`) instead of the default enum values (ie `0`,`1`,`2`,..`,n-1`) that easily correspond to indexes of an array.

Some solutions I can think of:
* Refactor `LogCategory` use the default enum values (ie `0`,`1`,...) and create another helper function (ie `LogCategoryToFlag(flag){ return 1 << flag;}` 
* Create a custom hash function for `LogCategoryToStr` `unordered_map` that returns a unique hash value for each enum value. This will guarantee that the `find()` will execute in constant time.
  *  Hash function could possibly be `flag % SOME_PRIME`.

However, I am not too sure if this additional complexity is worth it as the size of `LogCategoryToStr` is bounded

Reply:
Resolved the nit 👍🏻

Reply:
I guess at some point this might be refactored but doesn't necessarily need to be done here.